components/script/dom/globalscope.rs

```diff
@@ -2128,6 +2131,17 @@ impl GlobalScope {
             .borrow_mut()
             .create_pipeline_layout_id(backend)
     }
+
+    pub fn wgpu_create_shader_module_id(&self, backend: Backend) -> ShaderModuleId {
```

Comment:
nit: perhaps it would be simpler to just have a function returning `&mut GpuIdHub` and let the users call the methods appropriately?